contrib/fsl_integration/sct_plugin.py

```diff
@@ -208,52 +290,703 @@ def __init__(self, parent):
         button_gm = wx.Button(self, id=wx.ID_ANY,
                               label="Gray Matter Segmentation")
         button_gm.Bind(wx.EVT_BUTTON, self.onButtonGM)
+
+        hbox1 = wx.BoxSizer(wx.HORIZONTAL)
+        l1 = wx.StaticText(self, wx.ID_ANY, "Output File Name:")
+        hbox1.Add(l1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+        self.t1 = wx.TextCtrl(self)
+        self.t1.Bind(wx.EVT_TEXT, self.OnKeyTyped)
+        hbox1.Add(self.t1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+
         sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(hbox1)
         sizer.Add(button_gm, 0, wx.ALL, 5)
         self.sizer_h.Add(sizer)
         self.SetSizerAndFit(self.sizer_h)
 
+    def OnKeyTyped(self, event):
+        txt = event.GetString()
+
     def onButtonGM(self, event):
-        selected_overlay = displayCtx.getSelectedOverlay()
-        filename_path = selected_overlay.dataSource
-        cmd_line = "sct_deepseg_gm -i {} -o seg.nii.gz".format(filename_path)
+
+        overlayORD = displayCtx.overlayOrder
+
+        img1 = overlayORD[0]
+        rawimg = overlayList[img1].dataSource
+        gmimg = self.t1.GetValue()
+
+        print('Raw Image:', rawimg)
+
+        if gmimg == '':
+            print('Output file name not defined, standard name will be used.')
+            base_name = os.path.basename(rawimg)
+            fname, fext = base_name.split(os.extsep, 1)
+            gmname = "{}_gmseg.{}".format(fname, fext)
+            print('Output File Name:', gmname)
+
+            #Standard output file name
+            cmd_line = "sct_deepseg_gm -i {}".format(rawimg)
+            print('Command Line:', cmd_line)
+
+        else:
+            gmname = "{}.nii.gz".format(gmimg)
+            print('Output File Name:', gmname)
+
+            #Personalized output file name
+            cmd_line = "sct_deepseg_gm -i {} -o {}".format(rawimg, gmname)
+            print('Command Line:', cmd_line)
 
         self.call_sct_command(cmd_line)
 
-        outfilename = os.path.join(os.getcwd(), 'seg.nii.gz')
+        outfilename = os.path.join(os.getcwd(), gmname)
         image = Image(outfilename)
         overlayList.append(image)
-
         opts = displayCtx.getOpts(image)
         opts.cmap = 'yellow'
 
+#Computes Cross-Sectional Area for GM, WM and GM+WM (Total)
+class TabPanelCSA(SCTPanel):
+    DESCRIPTION = """This tool computes the spinal cord cross-sectional area. 
+    For more information, please refer to the article below.<br><br>
+    <b>Specific citation</b>:<br>
+    <b>Usage</b>:<br>
+    To launch the script, upload there are two aways to do that. First, upload the raw imaging (T1w, T2w or T2s) into FSLeyes,
```

Comment:
overcomplicated. why not simply say:
The input image (segmentation) should be at the top of the list.